Panic when trying to create with host

```shell
$ ghq create "git.p1.rocks/mkt/component-lib"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x7a055b]

goroutine 1 [running]:
main.doCreate(0xc000024c00, 0x2, 0x2)
	/home/runner/work/ghq/ghq/cmd_create.go:48 +0x24b
github.com/urfave/cli/v2.(*Command).Run(0xbdffc0, 0xc000024b00, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/command.go:161 +0x4e0
github.com/urfave/cli/v2.(*App).RunContext(0xc000001200, 0x9142e0, 0xc0000180f0, 0xc000010180, 0x3, 0x3, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:302 +0x814
github.com/urfave/cli/v2.(*App).Run(...)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:211
main.main()
	/home/runner/work/ghq/ghq/main.go:16 +0x67
```

--- a/cmd_create.go
+++ b/cmd_create.go
@@ -32,9 +32,6 @@ func doCreate(c *cli.Context) error {
 	if !ok {
 		return fmt.Errorf("directory %q already exists and not empty", p)
 	}
-	if err := os.MkdirAll(p, 0755); err != nil {
-		return err
-	}
 
 	remoteRepo, err := NewRemoteRepository(u)
 	if err != nil {
@@ -45,6 +42,14 @@ func doCreate(c *cli.Context) error {
 	if !ok {
 		vcsBackend, u = remoteRepo.VCS()
 	}
+	if vcsBackend == nil {
+		return fmt.Errorf("failed to init: unsupported VCS")
+	}
+
+	if err := os.MkdirAll(p, 0755); err != nil {
+		return err
+	}
+
 	initFunc := vcsBackend.Init
 	if initFunc == nil {
 		return fmt.Errorf("failed to init: unsupported VCS")

--- a/cmd_create.go
+++ b/cmd_create.go
@@ -46,14 +46,15 @@ func doCreate(c *cli.Context) error {
 		return fmt.Errorf("failed to init: unsupported VCS")
 	}
 
-	if err := os.MkdirAll(p, 0755); err != nil {
-		return err
-	}
-
 	initFunc := vcsBackend.Init
 	if initFunc == nil {
 		return fmt.Errorf("failed to init: unsupported VCS")
 	}
+
+	if err := os.MkdirAll(p, 0755); err != nil {
+		return err
+	}
+
 	if err := initFunc(p); err != nil {
 		return err
 	}

--- a/cmd_create_test.go
+++ b/cmd_create_test.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"errors"
 	"os"
 	"os/exec"
 	"path/filepath"
@@ -18,7 +19,7 @@ func TestDoCreate(t *testing.T) {
 		cmdutil.CommandRunner = orig
 	}(cmdutil.CommandRunner)
 	var lastCmd *exec.Cmd
-	cmdutil.CommandRunner = func(cmd *exec.Cmd) error {
+	commandRunner := func(cmd *exec.Cmd) error {
 		lastCmd = cmd
 		return nil
 	}
@@ -39,6 +40,7 @@ func TestDoCreate(t *testing.T) {
 		wantDir   string
 		errStr    string
 		setup     func() func()
+		cmdRun    func(cmd *exec.Cmd) error
 		skipOnWin bool
 	}{{
 		name:    "simple",
@@ -55,9 +57,13 @@ func TestDoCreate(t *testing.T) {
 		},
 		wantDir: filepath.Join(tmpd, "github.com/motemen/ghqqq"),
 	}, {
-		name:   "invalid VCS",
-		input:  []string{"create", "git-hub-git-hub-unknown.com/motemen/ghqqq"},
-		errStr: "invalid VCS",
+		name:  "invalid VCS",
+		input: []string{"create", "example.com/goooo/gooo"},
+		cmdRun: func(cmd *exec.Cmd) error {
+			lastCmd = cmd
+			return errors.New("bad repository")
+		},
+		errStr: "unsupported VCS",
 	}, {
 		name:    "Mercurial",
 		input:   []string{"create", "--vcs=hg", "motemen/ghq-hg"},
@@ -116,6 +122,11 @@ func TestDoCreate(t *testing.T) {
 				defer teardown()
 			}
 
+			cmdutil.CommandRunner = commandRunner
+			if tc.cmdRun != nil {
+				cmdutil.CommandRunner = tc.cmdRun
+			}
+
 			var err error
 			out, _, _ := capture(func() {
 				err = newApp().Run(append([]string{""}, tc.input...))
@@ -125,6 +136,7 @@ func TestDoCreate(t *testing.T) {
 			if tc.errStr == "" {
 				if err != nil {
 					t.Errorf("error should be nil, but: %s", err)
+					return
 				}
 			} else {
 				if err == nil {

--- a/cmd_create_test.go
+++ b/cmd_create_test.go
@@ -55,6 +55,10 @@ func TestDoCreate(t *testing.T) {
 		},
 		wantDir: filepath.Join(tmpd, "github.com/motemen/ghqqq"),
 	}, {
+		name:   "invalid VCS",
+		input:  []string{"create", "git-hub-git-hub-unknown.com/motemen/ghqqq"},
+		errStr: "invalid VCS",
+	}, {
 		name:    "Mercurial",
 		input:   []string{"create", "--vcs=hg", "motemen/ghq-hg"},
 		want:    []string{"hg", "init"},
@@ -123,6 +127,10 @@ func TestDoCreate(t *testing.T) {
 					t.Errorf("error should be nil, but: %s", err)
 				}
 			} else {
+				if err == nil {
+					t.Errorf("err should not be nil")
+					return
+				}
 				if e, g := tc.errStr, err.Error(); !strings.Contains(g, e) {
 					t.Errorf("err.Error() should contains %q, but not: %q", e, g)
 				}

--- a/cmd_get.go
+++ b/cmd_get.go
@@ -12,9 +12,9 @@ import (
 	"sync"
 
 	"github.com/mattn/go-isatty"
+	"github.com/urfave/cli/v2"
 	"github.com/x-motemen/ghq/cmdutil"
 	"github.com/x-motemen/ghq/logger"
-	"github.com/urfave/cli/v2"
 	"golang.org/x/sync/errgroup"
 )
 

--- a/local_repository.go
+++ b/local_repository.go
@@ -10,8 +10,8 @@ import (
 	"sync"
 
 	"github.com/Songmu/gitconfig"
-	"github.com/x-motemen/ghq/logger"
 	"github.com/saracen/walker"
+	"github.com/x-motemen/ghq/logger"
 )
 
 const envGhqRoot = "GHQ_ROOT"

--- a/main.go
+++ b/main.go
@@ -4,8 +4,8 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/x-motemen/ghq/logger"
 	"github.com/urfave/cli/v2"
+	"github.com/x-motemen/ghq/logger"
 )
 
 const version = "1.1.0"

